.github/workflows/ci.yml

```
@@ -0,0 +1,76 @@
+name: ci
+
+on:
+  pull_request: ~
+  push:
+    branches:
+      - main
+
+jobs:
+
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          cache: npm
+          node-version-file: .nvmrc
+      - run: npm ci
+      - run: npm run-script lint
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          cache: npm
+          node-version-file: .nvmrc
+      - run: npm ci
+      - run: npm run build
+      - uses: actions/upload-artifact@v3
+        with:
+          name: dist
+          path: dist
+          if-no-files-found: error
+
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          cache: npm
+          node-version-file: .nvmrc
+      - run: npm ci
+      - run: npm run test:unit -- --ci --reporters=default --reporters=jest-junit
+      - uses: actions/upload-artifact@v3
+        with:
+          name: test-results
+          path: "**/junit.xml"
```

NIT: just in case there are not test results, then report a failure?

```suggestion
          path: "**/junit.xml"
          if-no-files-found: error
```